app/src/main/java/nodomain/freeyourgadget/gadgetbridge/externalevents/BluetoothPairingRequestReceiver.java

```
@@ -0,0 +1,73 @@
+package nodomain.freeyourgadget.gadgetbridge.externalevents;
+
+import android.bluetooth.BluetoothDevice;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import nodomain.freeyourgadget.gadgetbridge.devices.DeviceCoordinator;
+import nodomain.freeyourgadget.gadgetbridge.impl.GBDevice;
+import nodomain.freeyourgadget.gadgetbridge.impl.GBDeviceCandidate;
+import nodomain.freeyourgadget.gadgetbridge.service.DeviceCommunicationService;
+import nodomain.freeyourgadget.gadgetbridge.util.DeviceHelper;
+
+/**
+ * Created by jpbarraca on 13/04/2017.
+ */
+
+public class BluetoothPairingRequestReceiver extends BroadcastReceiver {
+
+
+    private static final Logger LOG = LoggerFactory.getLogger(BluetoothConnectReceiver.class);
+
+    final DeviceCommunicationService service;
+
+    public BluetoothPairingRequestReceiver(DeviceCommunicationService service) {
+        this.service = service;
+    }
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        String action = intent.getAction();
+
+
+        if (!action.equals(BluetoothDevice.ACTION_PAIRING_REQUEST)) {
+            return;
+        }
+
+        GBDevice gbDevice = service.getGBDevice();
+        BluetoothDevice device = intent.getParcelableExtra(BluetoothDevice.EXTRA_DEVICE);
+        if (gbDevice == null || device == null)
+            return;
+
+        DeviceCoordinator coordinator = DeviceHelper.getInstance().getCoordinator(gbDevice);
+        GBDeviceCandidate candidate = new GBDeviceCandidate(device, gbDevice.getRssi(), device.getUuids());
+        try {
+            if (coordinator.getBondingStyle(candidate) == DeviceCoordinator.BONDING_STYLE_NONE) {
+                LOG.info("Aborting unwanted pairing request");
+                abortBroadcast();
+            }
+        } catch (Exception e) {
+
```

Exceptions should really never be swallowed. Just log them if you don't want to inform the user.